subprojects/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/NormalizingGroovyCompiler.java

```diff
@@ -76,15 +85,22 @@ public boolean apply(File element) {
         spec.setSourceFiles(ImmutableSet.copyOf(filtered));
     }
 
-    private void resolveClasspath(GroovyJavaJointCompileSpec spec) {
+    private List<File> resolveClasspath(GroovyJavaJointCompileSpec spec) {
```

Comment:
Can we keep the `void` return type here and store the `compileClasspath` of the spec in `withExtraClasspath`? I think it is confusing for this method to return the original classpath and not the resolved one.

Reply:
Done.